readthedocs/doc_builder/python_environments.py

```diff
@@ -72,6 +74,112 @@ def delete_existing_venv_dir(self):
             )
             shutil.rmtree(venv_dir)
 
+    def install_build_tools(self):
+        if settings.RTD_DOCKER_COMPOSE:
+            # Create a symlink for ``root`` user to use the same ``.asdf``
+            # installation than ``docs`` user. Required for local building
+            # since everything is run as ``root`` when using Local Development
+            # instance
+            cmd = [
+                'ln',
+                '-s',
+                os.path.join(settings.RTD_DOCKER_WORKDIR, '.asdf'),
+                '/root/.asdf',
+            ]
+            self.build_env.run(
+                *cmd,
+            )
+
+        for tool, version in self.config.build.tools.items():
+            version = version.full_version  # e.g. 3.9 -> 3.9.7
+
+            # TODO: generate the correct path for the Python version
+            # tool_path = f'{self.config.build.os}/{tool}/2021-08-30/{version}.tar.gz'
```

Comment:
I don't understand this `TODO`. Are the Python versions stored somewhere differently?

Reply:
I thought we could prefix the .tar.gz files with a date, so we can upgrade them (compile and upload to S3) without breaking production. I wrote about this in https://github.com/readthedocs/readthedocs.org/pull/8447#issuecomment-911562267 but we didn't discuss it yet; so I added a TODO here to come back.

Reply:
I'd link to that comment, it provides much more actionable context 👍